pkg/skaffold/walk/walk.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package walk
+
+import (
+	"os"
+
+	"github.com/karrick/godirwalk"
+)
+
+type Dirent interface {
+	IsDir() bool
+	Name() string
+}
+
+type Predicate func(string, Dirent) bool
+
+type Action func(string, Dirent) error
+
+type Excluder interface {
+	Exclude(relPath string) (bool, error)
+	SkipDir(relPath string) bool
+}
+
+type Builder interface {
+	// Options
+	Unsorted() Builder
+
+	// Predicates
+	When(Predicate) Builder
+	WhenIsDir() Builder
+	WhenIsFile() Builder
+	WhenHasName(string) Builder
+
+	// Actions
+	Do(Action) error
+	MustDo(Action)
+	AppendPaths(*[]string) error
+	CollectPaths() ([]string, error)
+}
```

Comment:
Wouldn't it make sense to incorporate `Excluder` in the `Builder` interface?

Reply:
yes, I will in a later PR